File: frontend/src/types/digest.ts
export interface DigestSection {
  theme: string;
  title: string;
  body: string;
  article_ids: number[];
}

export interface DigestContent {
  headline: string;
  sections: DigestSection[];
  key_takeaways: string[];
  connections: string;
}

export interface Digest {
  id: number;
  digest_date: string;
  content: DigestContent;
  article_ids: number[];
  article_count: number;
  created_at: string;
  updated_at: string;
}

File: backend/schemas/digest.py
"""Digest response schemas."""

from datetime import date, datetime

from pydantic import BaseModel


class DigestSectionResponse(BaseModel):
    """Single thematic section within a digest."""

    theme: str
    title: str
    body: str
    article_ids: list[int]


class DigestContentResponse(BaseModel):
    """Structured digest content."""

    headline: str
    sections: list[DigestSectionResponse]
    key_takeaways: list[str]
    connections: str


class DigestResponse(BaseModel):
    """Full digest API response."""

    id: int
    digest_date: date
    content: DigestContentResponse
    article_ids: list[int]
    article_count: int
    created_at: datetime
    updated_at: datetime
